File: WebShop/WebShop-API/DTOs/Product/DirectProductResponse.cs
namespace WebShop_API.DTOs.Product
{
    public class DirectProductResponse
    {
        public int ProductID { get; set; } = 0;

        public string ProductName { get; set; } = string.Empty;

        public double ProductPrice { get; set; } = 0;

        public int ProductQuantity { get; set; } = 0;

        public string ProductDescription { get; set; } = string.Empty;

        public StaticManufacturerResponse Manufacturer { get; set; } = null!;

        public StaticCategoryResponse Category { get; set; } = null!;

        public DateTime ReleaseDate { get; set; }

        public StaticProductTypeResponse ProductType { get; set; } = null!;

        public StaticDiscountResponse? Discount { get; set; }

        public List<StaticTransactionResponse> Transactions { get; set; } = new();

        public List<StaticPhotoResponse> Photos { get; set; } = new();
    }
}

File: WebShop/WebShop-Client/src/app/_models/product/DirectProductResponse.ts
import { StaticCategoryResponse } from "../category";
import { StaticDiscountResponse } from "../discount";
import { StaticManufacturerResponse } from "../Manufacturer";
import { StaticPhotoResponse } from "../photo";
import { StaticProductTypeResponse } from "../productType";
import { StaticTransactionResponse } from "../transaction";

export interface DirectProductResponse
{
	productID: number;
	productName: string;
	productPrice: number;
	productQuantity: number;
	productDescription: string;
	manufacturer: StaticManufacturerResponse;
	category: StaticCategoryResponse;
	releaseDate: string;
	productType: StaticProductTypeResponse;
	discount: StaticDiscountResponse | null;
	transactions: StaticTransactionResponse[];
	photos: StaticPhotoResponse[];
}
